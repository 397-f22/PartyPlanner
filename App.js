import "./App.css";
import React, { useState } from "react";
import RestaurantList from "./components/RestaurantList.js";
import FilterMenu from "./components/FilterMenu.js";
import NavBar from "./components/NavBar.js";
import SortBar from "./components/SortBar";
import RestaurantData from "./Restaurants.json";

import { useData } from "./utilities/firebase.js";
import "bootstrap/dist/css/bootstrap.min.css";



const App = () => {
  // const [sortWage, setSortWage] = useState("")
  const [sortDirection, setSortDirection] = useState("");
  const [filterCategories, setFilterCategories] = useState([]);
  //const [RestaurantData, loading, error] = useData("/");

  //if (error) return <h1>{error}</h1>;
  //if (loading) return <h1>Loading your restaurants...</h1>;

  const restaurantCategories = Array.from(
    new Set(
      Object.values(RestaurantData).map((rest) => {
        return rest.CATEGORY;
      })
    )
  );

  return (
    <div className="app-body">
      <NavBar />
      <div className="container">
        <div className="filters-and-restaurants">
          <div className="filters-list">
            <FilterMenu
              restaurantCategories={restaurantCategories}
              filterCategories={filterCategories}
              setFilterCategories={setFilterCategories}
              sortDirection={sortDirection}
              setSortDirection={setSortDirection}
            />
          </div>
          <div className="cards-list">
            <RestaurantList
              restaurants={RestaurantData}
              filterCategories={filterCategories}
              sortDirection={sortDirection}
            />
          </div>
        </div>
      </div>
    </div>
  );
};

export default App;
