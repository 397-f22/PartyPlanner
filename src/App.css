
.App {
  text-align: center;
}

.app-body * {
  box-sizing: border-box;
  /* border: solid 2px black; */
}

html, .app-body {
  display: flex;
  flex-direction: column;
  background-color: #e7ecef;
  height: 100%;
}



Nav {
  top: 0;
  position: sticky;
  z-index: 100;
}

.app-content {
  display: flex;
  flex-direction: column;
}

.container {
  display: flex;
  flex-direction: column;
}

.sort-bar{
  align-self: flex-end;
}

.filters-and-jobs{
  display: flex;
  flex-direction: column;
}

.cards-list{
  /* width: 80%; */
}

.filters-list{
  top: 85px;
  z-index: 999;
}