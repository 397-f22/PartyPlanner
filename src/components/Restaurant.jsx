
import React, { useState } from 'react';
import '../styles/Restaurant.css'
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';
import Modal from '@mui/material/Modal';

const openLink = (url) => {
    window.open(url)
}

const style = {
    position: 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
    width: 800,
    bgcolor: 'background.paper',
    border: '2px solid #000',
    boxShadow: 24,
    p: 4,

};

const Restaurant = ({ restaurant }) => {

    return (
        <div className='container-fluid' data-testid='restaurant'>
            <div className="card" data-testid={`restaurant-${restaurant.TITLE.substring(restaurant.TITLE.indexOf(":") + 1)}`}>
                <div className="card-title"> {restaurant.TITLE.substring(restaurant.TITLE.indexOf(":") + 1)}
                </div>
                <div className="card-category">
                    {restaurant.CUSINE}
                </div>

                <div className="card-body">
                    <div className="row">
                        <div className='card-text col-sm'> ${restaurant["PRICE"][0]} </div>
                        <div className='card-text col-sm'> {restaurant["AVAILABILITY"]} </div>
                        <div className='card-text col-sm'> {restaurant.LOCATION} </div>
                    </div>
                </div>
            </div>
        </div>
    )
}

export default Restaurant