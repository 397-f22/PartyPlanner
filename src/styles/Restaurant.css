.cards {
    width: 100%;
    padding-left: 0;
}

.container-fluid{
    padding-left: 0;
}
.card {
    background-color: #e7ecef;
    margin-top: 15px;
}

.card-category {
    padding-left: 10px;
    display: flex;
    align-items: center;
}

.card-text {
    color: #7a7a7a;
}

.card-title {
    font-weight: bold;
    padding-top: 10px;
    padding-left: 10px;
    color: #d8a47f;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}


