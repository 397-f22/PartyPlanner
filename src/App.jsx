import "./App.css";
import React, { useState } from "react";
import RestaurantList from "./components/RestaurantList.jsx";
import FilterMenu from "./components/FilterMenu.jsx";
import NavBar from "./components/NavBar.jsx";
import SortBar from "./components/SortBar";
import RestaurantData from "./Restaurants.json";
import Button from '@mui/material/Button';
import Modal from '@mui/material/Modal';
import Box from '@mui/material/Modal';
//import { useData } from "./utilities/firebase.js";
import "bootstrap/dist/css/bootstrap.min.css";
import FilterIcon from "./pics/filter.png";



const App = () => {
  // const [sortWage, setSortWage] = useState("")
  const [sortDirection, setSortDirection] = useState("");
  const [selectedOptions, setSelectedOptions] = useState({"Dietary Restrictions": [], "Group Size": null, "Price Level": null});
  //const [RestaurantData, loading, error] = useData("/");

  //if (error) return <h1>{error}</h1>;
  //if (loading) return <h1>Loading your restaurants...</h1>;
  
  // SAVE ALL POSSIBLE FILTERING CATEGORIES IN ONE VARIABLE
  let allOptions = {}
  // Get all dietary restriction categories
  let dietaryRestrictions = []
  Object.values(RestaurantData).map((rest) => {
      rest.CATEGORY.map(cat => dietaryRestrictions.push(cat));
    })
  dietaryRestrictions = new Array(...new Set(dietaryRestrictions));
  // Get all group size options
  let groupSize = [5,20]
  // Get all price level options
  let priceLevels = []
  Object.values(RestaurantData).map((rest) => {
    priceLevels.push(rest.PRICE[0]);
  })
  priceLevels = new Array(...new Set(priceLevels));
  console.log(priceLevels);
  // Add all filtering options into allFilterCategories
  allOptions["Dietary Restrictions"] = dietaryRestrictions.sort();
  allOptions["Group Size"] = groupSize;
  allOptions["Price Level"] = priceLevels.sort();
  const [show,setShow] = useState(false)
  const handleOpen = () => {
    setShow(true)
  };
  const handleClose = () => {
    setShow(false)
  };
  const filterButtonStyle = {
    marginLeft: "12px",
    marginTop: "15px",
    backgroundColor: "#d8a47f",
    color: "black",
    border: "1px solid #d8a47f",
  }
 
  return (
    <div className="app-body">
      <NavBar />
      <div className="container">
        <div className="filters-and-restaurants">
          <div className="filters-list">
          
          <Button sx={filterButtonStyle} onClick={handleOpen} startIcon={<img style={{width: "20p", height: "20px", display:"block" }} src={FilterIcon}></img>} >Filter Options</Button>

          <Modal
            open={show}
            onClose={handleClose}
            aria-labelledby="modal-modal-title"
            aria-describedby="modal-modal-description"
          >
            {/* <Box sx={style}>
              <div>test</div>
            </Box> */}
            <FilterMenu
              allOptions={allOptions}
              selectedOptions={selectedOptions}
              setSelectedOptions={setSelectedOptions}
              sortDirection={sortDirection}
              setSortDirection={setSortDirection}
            />
          </Modal>
          </div>
          <div className="cards-list">
            <RestaurantList
              restaurants={RestaurantData}
              selectedOptions={selectedOptions}
              sortDirection={sortDirection}
            />
          </div>
        </div>
      </div>
    </div>
  );
};

export default App;
