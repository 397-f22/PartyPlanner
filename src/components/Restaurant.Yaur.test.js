import { getByTestId, render, screen } from '@testing-library/react';
import App from '../App';
import userEvent from '@testing-library/user-event';
import { async } from '@firebase/util';

import Restaurant from './Restaurant';

test('restaurant should have name, price, and location', ()=> {
    const fakeRestaurant = {
        "CATEGORY": ["Gluten free", "Nut Free", "Vegetarian"],
        "NUMBER": "847-866-9900",
        "CUSINE": "Spanish",
        "ID": 6,
        "LOCATION": "1615 Chicago Ave, Evanston, IL 60201",
        "PRICE": [
          "$"
        ],
        "TITLE": "Tapas Barcelona",
        "AVAILABILITY": "Opens 5PM",
        "CAPACITY": [0,0,0,0,0,0,0,0,5,7,9,12,11,15,11,13,8,11,6,9,10,11,0,0],
        "URL": "https://www.allmenus.com/il/evanston/16743-tapas-barcelona/menu/"
      }
    const { getByTestId,debug } = render(<Restaurant restaurant={fakeRestaurant} />); 
    expect(getByTestId('name').textContent).toContain("Tapas Barcelona")
    expect(getByTestId('price').textContent).toContain("$")
    expect(getByTestId('location').textContent).toContain("1615 Chicago Ave, Evanston, IL 60201")
})