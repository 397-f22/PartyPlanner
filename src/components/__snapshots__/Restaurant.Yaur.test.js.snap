// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`restaurant should have name, price, and location 1`] = `
<DocumentFragment>
  <div
    class="container-fluid"
    data-testid="restaurant"
  >
    <div
      class="card"
      data-testid="restaurant-Tapas Barcelona"
    >
      <div
        class="card-top"
      >
        <div
          class="card-title"
          data-testid="name"
        >
           Tapas Barcelona
        </div>
        <div
          class="view-menu"
        >
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1ectwcq-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            View Menu
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root css-178nucd-MuiButtonBase-root"
            tabindex="0"
            type="button"
          >
            <span
              class="material-symbols-outlined"
              id="call-button"
            >
              smartphone
            </span>
          </button>
        </div>
      </div>
      <div
        class="card-category"
      >
        Spanish
      </div>
      <div
        class="card-body"
      >
        <div
          class="row"
        >
          <div
            class="card-text col-sm"
            data-testid="price"
          >
             $ 
          </div>
          <div
            class="card-text col-sm"
          >
             Opens 5PM 
          </div>
          <div
            class="card-text col-sm"
            data-testid="location"
          >
             1615 Chicago Ave, Evanston, IL 60201 
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
